client/src/selectors/index.js

```diff
@@ -70,4 +71,66 @@ export const mapOptionsSelector = (state: State): MapOptions => state.mapOptions
 export const centerSelector = (state: State): [number, number] => state.mapOptions.center
 export const zoomSelector = (state: State): number => state.mapOptions.zoom
 export const mapReferenceSelector = (state: State): MapReference => state.mapReference
-export const entitiesSelector = (state: State): Array<Entity> => state.entities
+export const entitiesSelector = (state: State): ?Array<Entity> => state.entities
+export const currentPageSelector = (state: State): number => state.publicly.currentPage
+
+export const entitiesLengthSelector = createSelector(
+  entitiesSelector,
+  (entities) => (entities ? entities.length : 0)
+)
+
+export const pageCountSelector = createSelector(
+  entitiesSelector,
+  (entities) => (entities ? Math.ceil(entities.length / VEREJNE_MAX_PAGE_ITEMS) : 0)
+)
+
+// TODO performance
```

Comment:
an easy performance save is to make a selector for entities sorted by size, and then use this one just to chunk and paginate